mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulFollowingScreen.kt

```diff
@@ -11,42 +12,57 @@ import ch.epfl.sdp.mobile.ui.social.SocialScreen
 import ch.epfl.sdp.mobile.ui.social.SocialScreenState
 import ch.epfl.sdp.mobile.ui.social.SocialScreenState.Mode.Following
 import ch.epfl.sdp.mobile.ui.social.SocialScreenState.Mode.Searching
+import kotlinx.coroutines.flow.*
 
-private class SnapshotSocialScreenState(
-    private val following: State<List<Profile>>,
-) : SocialScreenState {
-
-  private data class ProfileAdapter(
-      private val profile: Profile,
-  ) : Person {
-    override val backgroundColor: Color
-      get() = profile.backgroundColor
-    override val name: String
-      get() = profile.name
-    override val emoji: String
-      get() = profile.emoji
-  }
-
-  override var mode: SocialScreenState.Mode by mutableStateOf(Following)
-
-  override var players: List<Person> = following.value.map { ProfileAdapter(it) }
-  override var input: String by mutableStateOf("")
-
-  // TODO :  define how to return to the following screen, return button ?
-  // TODO :  Modify here to update the player list when we do a search
-  override fun onValueChange() {
-    if (mode != Following) {
-      mode = Searching
-    }
-  }
+private data class ProfileAdapter(
+    private val profile: Profile,
+) : Person {
+  override val backgroundColor: Color
+    get() = profile.backgroundColor
+  override val name: String
+    get() = profile.name
+  override val emoji: String
+    get() = profile.emoji
 }
 
 @Composable
 fun StatefulFollowingScreen(
     user: AuthenticatedUser,
     modifier: Modifier = Modifier,
 ) {
-  val state = remember(user) { user.following }.collectAsState(emptyList())
+  val following =
+      remember(user) { user.following }.collectAsState(emptyList()).value.map { ProfileAdapter(it) }
+
+  val socialFacade = LocalSocialFacade.current
+  val input = remember { mutableStateOf("") }
+  val searchResults =
+      remember { snapshotFlow { input.value }.flatMapLatest { s -> socialFacade.search(s) } }
+          .collectAsState(emptyList())
+          .value
+          .map { ProfileAdapter(it) }
+
+  val searchFieldInteraction = remember { MutableInteractionSource() }
+  val focused by searchFieldInteraction.collectIsFocusedAsState()
+  val mode = if (focused) Searching else Following
+
+  SocialScreen(
+      SnapshotSocialScreenState(following, input, searchResults, mode, searchFieldInteraction),
+      modifier)
+}
+
+private class SnapshotSocialScreenState(
+    following: List<Person>,
+    input: MutableState<String>,
+    searchResult: List<Person>,
+    mode: SocialScreenState.Mode,
+    searchFieldInteraction: MutableInteractionSource
+) : SocialScreenState {
+  override var following = following
+  override var input by input
+  override var searchResult = searchResult
+  override var mode = mode
+  override var searchFieldInteraction = searchFieldInteraction
```

Comment:
Nitpick : if most the fields set, you maybe could create a `data class` with a syntax as such :

```kotlin
data class SnapshotSocialScreenState(override val following: Boolean): SocialScreenState
```

Also, please prefer using `val` when you can.